[BUG] typescript generator throws when `additionalProperties` is set to `true`

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

When using the `typescript` generator, anywhere `additionalProperties` is explicitly set to `true`, the generator throws. It should be expected with array types, but this happens on object types. This is not the case with the `typescript-axios` or `typescript-fetch` generators. It does not throw when `additionalProperties` is set to `false` or if it's omitted (in which case I've interpreted the spec as to it's true by default).

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->

6.0.1

##### OpenAPI declaration file content or url
<details>
<summary>Repro JSON schema</summary>
<p>

```json
{
    "openapi": "3.0.3",
    "components": {
        "schemas": {
            "foo": {
                "properties": {
                    "bar": {
                        "additionalProperties": true,
                        "description": "baz",
                        "type": "object"
                    }
                },
                "type": "object"
            }
        }
    },
    "info": {
        "title": "Repro",
        "version": "1.0.0"
    },
    "paths": {
        "/": {
            "get": {
                "description": "Foo",
                "operationId": "getFoo",
                "responses": {
                    "200": {
                        "content": {
                            "application/json": {
                                "schema": {
                                    "properties": {
                                        "foo": {
                                            "description": "Foo.",
                                            "type": "string"
                                        }
                                    },
                                    "required": [
                                        "foo"
                                    ],
                                    "type": "object"
                                }
                            }
                        },
                        "description": "Foo"
                    }
                },
                "summary": "Foo",
                "tags": [
                    "metadata"
                ]
            }
        }
    }
}

```

</p></details>

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->

```bash
openapi-generator-cli generate -i spec.json -g typescript -o /dev/null
```

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->

Use the `typescript` generator to generate code from any schema where any `components.schemas.properties` object has `additionalProperties` set to `true`.

<details>
<summary>Logs</summary>
<p>

```bash
Exception in thread "main" java.lang.RuntimeException: Could not process model 'errorAuthenticatorAssuranceLevelNotSatisfied'.Please make sure that your schema is correct!
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:509)
        at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:911)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:465)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class io.swagger.v3.oas.models.media.Schema (java.lang.Boolean is in module java.base of loader 'bootstrap'; io.swagger.v3.oas.models.media.Schema is in unnamed module of loader 'app')
        at org.openapitools.codegen.languages.TypeScriptClientCodegen.getTypeDeclaration(TypeScriptClientCodegen.java:890)
        at org.openapitools.codegen.DefaultCodegen.fromProperty(DefaultCodegen.java:3650)
        at org.openapitools.codegen.DefaultCodegen.addVars(DefaultCodegen.java:5402)
        at org.openapitools.codegen.DefaultCodegen.addVars(DefaultCodegen.java:5344)
        at org.openapitools.codegen.DefaultCodegen.updateModelForObject(DefaultCodegen.java:2672)
        at org.openapitools.codegen.DefaultCodegen.fromModel(DefaultCodegen.java:2873)
        at org.openapitools.codegen.DefaultGenerator.processModels(DefaultGenerator.java:1288)
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:504)
        ... 4 more
```

</p>
</details>

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java
@@ -887,7 +887,7 @@ public class TypeScriptClientCodegen extends DefaultCodegen implements CodegenCo
             inner = ((ArraySchema) p).getItems();
             return this.getSchemaType(p) + "<" + this.getTypeDeclaration(unaliasSchema(inner)) + ">";
         } else if (ModelUtils.isMapSchema(p)) {
-            inner = (Schema) p.getAdditionalProperties();
+            inner = getSchemaAdditionalProperties(p);
             return "{ [key: string]: " + this.getTypeDeclaration(unaliasSchema(inner)) + "; }";
         } else if (ModelUtils.isFileSchema(p)) {
             return "HttpFile";

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/TypeScriptClientCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/TypeScriptClientCodegenTest.java
@@ -79,4 +79,23 @@ public class TypeScriptClientCodegenTest {
         Assert.assertFalse(codegenModel.imports.contains("Set"));
     }
 
+    @Test
+    public void testWithAdditionalProperties() {
+        final Schema inner = new ObjectSchema();
+        inner.setAdditionalProperties(true);
+
+        final Schema root = new ObjectSchema()
+            .addProperties("inner", inner);
+
+        final DefaultCodegen codegen = new TypeScriptClientCodegen();
+        final OpenAPI openAPI = TestUtils.createOpenAPIWithOneSchema("sample", root);
+        codegen.setOpenAPI(openAPI);
+
+        try {
+            // TypeScriptClientCodegen can generate codes without any exception.
+            codegen.fromModel("sample", root);
+        } catch (Exception e) {
+            Assert.fail("Exception was thrown.");
+        }
+    }
 }

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/TypeScriptClientCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/TypeScriptClientCodegenTest.java
@@ -92,7 +92,7 @@ public class TypeScriptClientCodegenTest {
         codegen.setOpenAPI(openAPI);
 
         try {
-            // TypeScriptClientCodegen can generate codes without any exception.
+            // TypeScriptClientCodegen can generate codes without throwing exception.
             codegen.fromModel("sample", root);
         } catch (Exception e) {
             Assert.fail("Exception was thrown.");

